File: Arrays_2/character_picture_grid.py
import time

start_time = time.time()
grid = [
    [".", ".", ".", ".", ".", "."],
    [".", "O", "O", ".", ".", "."],
    ["O", "O", "O", "O", ".", "."],
    ["O", "O", "O", "O", "O", "."],
    [".", "O", "O", "O", "O", "O"],
    ["O", "O", "O", "O", "O", "."],
    ["O", "O", "O", "O", ".", "."],
    [".", "O", "O", ".", ".", "."],
    [".", ".", ".", ".", ".", "."],
]

width = len(grid[0])
height = len(grid)
result = []

for col in range(width):
    for row in grid:
        result.append(row[col])
    result.append("\n")

result_string = "".join(result)
end_time = time.time()
print(result_string)
elapsed_time = end_time - start_time
print(f"Time taken: {elapsed_time:.6f} seconds")

File: Arrays_2/src/bin/character_picture_grid.rs
use std::time::Instant;

fn main() {
    let start_time = Instant::now();
    let picture_grid = vec![
        vec!['.', '.', '.', '.', '.', '.'],
        vec!['.', 'O', 'O', '.', '.', '.'],
        vec!['O', 'O', 'O', 'O', '.', '.'],
        vec!['O', 'O', 'O', 'O', 'O', '.'],
        vec!['.', 'O', 'O', 'O', 'O', 'O'],
        vec!['O', 'O', 'O', 'O', 'O', '.'],
        vec!['O', 'O', 'O', 'O', '.', '.'],
        vec!['.', 'O', 'O', '.', '.', '.'],
        vec!['.', '.', '.', '.', '.', '.'],
    ];

    let width = picture_grid[0].len();
    let height = picture_grid.len();

    let mut result = String::with_capacity(width * (height + 1));

    for col in 0..width {
        for row in &picture_grid {
            result.push(row[col]);
        }
        result.push('\n');
    }
    let elapsed_time = start_time.elapsed();
    print!("{}", result);

    println!("Time taken: {:.6} seconds", elapsed_time.as_secs_f64());
}
